pkg/labels/labels_test.go

```diff
@@ -629,3 +630,30 @@ func TestBuilder_Labels(t *testing.T) {
 		}).Labels(),
 	)
 }
+
+func TestLabels_Hash(t *testing.T) {
+	lbls := Labels{
+		{Name: "foo", Value: "bar"},
+		{Name: "baz", Value: "qux"},
+	}
+	testutil.Equals(t, lbls.Hash(), lbls.Hash())
+	testutil.Assert(t, lbls.Hash() != Labels{lbls[1], lbls[0]}.Hash(), "unordered labels match.")
+	testutil.Assert(t, lbls.Hash() != Labels{lbls[0]}.Hash(), "different labels match.")
+}
+
+var benchmarkLabels_Result uint64
+
+func BenchmarkLabels_Hash(b *testing.B) {
+	b.ReportAllocs()
+
+	lbl := &strings.Builder{}
+	lbl.Grow(1024 * 1024 * 10) // 10MB.
+	word := "abcdefghij"
+	for i := 0; i < lbl.Cap()/len(word); i++ {
+		_, _ = lbl.WriteString(word)
+	}
+	lbls := Labels{{Name: "__name__", Value: lbl.String()}}
+	for i := 0; i < b.N; i++ {
```

Comment:
You need to reset allocs before the for to get the right values.

Reply:
It would be nice yes, but also will not matter here.